dashboard/app/models/pd/scholarship_info.rb

```diff
@@ -33,9 +33,9 @@ class Pd::ScholarshipInfo < ActiveRecord::Base
   belongs_to :enrollment, class_name: 'Pd::Enrollment', foreign_key: :pd_enrollment_id
   belongs_to :application, class_name: 'Pd::Application::TeacherApplicationBase', foreign_key: :pd_application_id
 
+  validate :course_specific_scholarship_status
```

Comment:
How about something like `:scholarship_must_be_valid_for_course`? It makes the meaning of the validation clearer.
Other examples: https://guides.rubyonrails.org/active_record_validations.html#custom-methods (although I think the function names in this link are quite verbose).

Reply:
Much better, thanks @hacodeorg! Will update.